### PR TITLE
[gym] Remove legacy build API

### PR DESCRIPTION
**Pull request by KrauseFx**
It's about time

### CODE REVIEW

**Comment by fastlane-bot-helper** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Big PR</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">Danger</a>
</p>

File: gym/lib/gym/generators/package_command_generator_xcode7.rb [hunk 1]
```diff
@@ -46,6 +44,12 @@ def temporary_output_path
         Gym.cache[:temporary_output_path] ||= Dir.mktmpdir('gym_output')
       end
 
+      # Wrap xcodebuild to work-around ipatool dependecy to system ruby
```
Comment (ohayon) [inline gym/lib/gym/generators/package_command_generator_xcode7.rb]: *dependency

**Review by taquitos — APPROVED** [review]
🆗 🙏 🐑 it!

**Review by ohayon — CHANGES_REQUESTED** [review]
This is so amazing. I just did a quick pass, and the one question I have other than the little typo is: should we show people a message that this has been removed?

**Comment by KrauseFx** [comment]
@ohayon Great question. Looking at the diff, we warned people angrily to stop using it for the last year, and since about 6 months or so, the legacy API doesn't work at all, so I think we should be fine

**Review by ohayon — APPROVED** [review]

**Comment by ohayon** [comment]
👍 Grrreat!

**Comment by KrauseFx** [comment]
Let's wait for a few days before merging and see if someone has a concern 👍

File: gym/lib/gym/options.rb [hunk 2]
```diff
@@ -200,15 +166,6 @@ def self.plain_options
                                      description: "The toolchain that should be used for building the application (e.g. com.apple.dt.toolchain.Swift_2_3, org.swift.30p620160816a)",
                                      optional: true,
                                      is_string: false),
-        FastlaneCore::ConfigItem.new(key: :provisioning_profile_path,
```
Comment (lacostej) [inline gym/lib/gym/options.rb]: Sorry for delay, this got stuck in github 'start a review' mode :)

Comment (KrauseFx) [inline gym/lib/gym/options.rb]: Yeah it will fail, _fastlane_ will tell the user that this option isn't available.

**Comment by KrauseFx** [comment]
Alright, should we merge this now? :)

**Comment by ohayon** [comment]
@KrauseFx YEAH! \o/

**Comment by ryanwiesjahn** [comment]
@KrauseFx So I was actually using the `provisioning_profile_path` option of `gym`. I knew it was deprecated, but I'm in a situation where I can't use `match` or `cert` and `sigh`, because those all seem to require admin access to the Apple Developer account which I do not have. What is the proper way to specify signing cert and provisioning profile files that I am providing to Fastlane?

**Comment by KrauseFx** [comment]
Sorry we probably broke your setup @ryanwiesjahn, it was necessary though, as Xcode doesn't support that option any more. Instead, please follow our [code signing guide](https://docs.fastlane.tools/codesigning/getting-started/), in particular the Xcode setup 👍 Please submit a new issue if you have any questions :)

**Pull request merged**